### PR TITLE
Ensure that metadynamics uses slightly different initial geometry

### PR DESCRIPTION
**Pull request by susilehtola**
See #802

### CODE REVIEW

File: src/dynamic.f90 [hunk 1]
```diff
@@ -342,10 +344,23 @@ subroutine md(env,mol,chk,calc, &
       endif
       if (metasetlocal%nstruc.eq.0) then
          do i = 1, mol%n
-            do j = 1, 3
-               call random_number(dum2)
-               metasetlocal%xyz(j,i,1) = mol%xyz(j,i) + 1.0e-6_wp*dum2
+            ! Generate randomly displaced geometry
+            do while(.true.)
```
Comment (awvwgk) [inline src/dynamic.f90:348]: Just ``do`` is sufficient here as it implies an always true logical expression


```suggestion
            do
```

File: src/dynamic.f90 [hunk 2]
```diff
@@ -342,10 +344,23 @@ subroutine md(env,mol,chk,calc, &
...
+               rnorm = 0.0_wp
+               do j = 1, 3
+                  call random_number(rcoord(j))
+                  rnorm = rnorm + rcoord(j)**2
+               enddo
+               ! Ensure that displacement is large enough so that it
+               ! can be normalized to the unit sphere
+               if(rnorm .ge. 1e-8) then
```
Comment (awvwgk) [inline src/dynamic.f90:356]:
```suggestion
               if(rnorm >= 1e-8) then
```

File: src/dynamic.f90 [hunk 3]
```diff
@@ -342,10 +344,23 @@ subroutine md(env,mol,chk,calc, &
...
       if (metasetlocal%nstruc.eq.0) then
          do i = 1, mol%n
-            do j = 1, 3
-               call random_number(dum2)
-               metasetlocal%xyz(j,i,1) = mol%xyz(j,i) + 1.0e-6_wp*dum2
+            ! Generate randomly displaced geometry
+            do while(.true.)
+               rnorm = 0.0_wp
+               do j = 1, 3
+                  call random_number(rcoord(j))
```
Comment (awvwgk) [inline src/dynamic.f90:351]: I think this short loop is unnecessary if you pass the whole `rcoord` array to `random_number`

Comment (susilehtola) [reply src/dynamic.f90]: True, there's also an intrinsinc function for computing the norm. My FORTRAN is a bit rusty

**Review by awvwgk — APPROVED** [review]
Thanks for sharing, looks good to me now.

File: src/dynamic.f90 [hunk 4]
```diff
@@ -342,10 +344,22 @@ subroutine md(env,mol,chk,calc, &
...
+               ! Ensure that displacement is large enough so that it
+               ! can be normalized to the unit sphere
+               if(norm2(rcoord) >= 1e-8) then
+                  exit
+               endif
             enddo
+            ! Normalize displacement to the unit sphere
+            rcoord = rcoord/norm2(rcoord)
+            ! Assign displaced geometry
+            metasetlocal%xyz(j,i,1) = mol%xyz(j,i) + atom_displacement*rcoord(j)
```
Comment (awvwgk) [inline src/dynamic.f90:362]: `j` is unbound here
```suggestion
            metasetlocal%xyz(:,i,1) = mol%xyz(:,i) + atom_displacement*rcoord
```

Comment (susilehtola) [reply src/dynamic.f90]: Thanks, forgot to add a loop

**Review by MtoLStoN — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**